test/get.js

```diff
@@ -1,29 +1,31 @@
 'use strict';
 
+const {deepStrictEqual: eq} = require ('assert');
 const vm = require ('vm');
 
 const $ = require ('sanctuary-def');
 
 const S = require ('..');
 
-const eq = require ('./internal/eq');
-
 
 test ('get', () => {
 
-  eq (String (S.get)) ('get :: (Any -> Boolean) -> String -> a -> Maybe b');
+  eq (String (S.get), 'get :: (Any -> Boolean) -> String -> a -> Maybe b');
+
+  eq (S.get (S.is ($.Number)) ('x') ({x: 0, y: 42}), S.Just (0));
+  eq (S.get (S.is ($.Number)) ('y') ({x: 0, y: 42}), S.Just (42));
+  eq (S.get (S.is ($.Number)) ('z') ({x: 0, y: 42}), S.Nothing);
+  eq (S.get (S.is ($.String)) ('z') ({x: 0, y: 42}), S.Nothing);
+  eq (S.get (S.is ($.String)) ('x') ({x: 0, y: 42}), S.Nothing);
 
-  eq (S.get (S.is ($.Number)) ('x') ({x: 0, y: 42})) (S.Just (0));
-  eq (S.get (S.is ($.Number)) ('y') ({x: 0, y: 42})) (S.Just (42));
-  eq (S.get (S.is ($.Number)) ('z') ({x: 0, y: 42})) (S.Nothing);
-  eq (S.get (S.is ($.String)) ('z') ({x: 0, y: 42})) (S.Nothing);
-  eq (S.get (S.is ($.String)) ('x') ({x: 0, y: 42})) (S.Nothing);
-  eq (S.get (S.is ($.RegExp)) ('x') ({x: vm.runInNewContext ('/.*/')})) (S.Just (/.*/));
+  // <https://github.com/nodejs/node/issues/44462>
+  const pattern = vm.runInNewContext ('/.*/');
+  eq (S.get (S.is ($.RegExp)) ('x') ({x: pattern}), S.Just (pattern));
```

Comment:
This change is necessary because [`assert.deepStrictEqual`][1] checks prototype identity (nodejs/node#44462). This assertion was testing three things at once: `S.is`, `S.get`, and our internal `eq` function. We never meant to assert that equivalent `RegExp` objects from different contexts are considered equal, as this has nothing to do with `S.get`.


[1]: https://nodejs.org/api/assert.html#assertdeepstrictequalactual-expected-message